proposals/0268-main-menu-updating.md

```diff
@@ -12,171 +12,118 @@ This proposal seeks to add the ability to update the main menu with artworks and
+<!-- New Struct -->
+<struct name="DynamicUpdateCapabilities" since="X.X">
+    <param name="supportedDynamicImageFieldNames" type="ImageFieldName" array="true" mandatory="false" minsize="1">
+        <description>An array of ImageFieldName values for which the system supports sending OnFileUpdate notifications. If you send an Image struct for that image field with a name without having uploaded the image data using PutFile that matches that name, the system will request that you upload the data with PutFile at a later point when the HMI needs it. The HMI will then display the image in the appropriate field. If not sent, assume false.</description>
     </param>
-</struct>
-```
-
-#### Dynamic Menu Updates
-As of this version of Core, sending an `AddCommand` or `AddSubMenu` with a `cmdID` that already exists **must not** be rejected but should update the existing command or submenu with the data from the new RPC. For `AddSubMenu` all existing commands that are attached to the submenu should be transferred.
-
-If data is not passed that previously existed, that data should be removed. e.g. If an `AddCommand` is sent with text and no image, then a new `AddCommand` (with the same `cmdID`) is sent with no text and an image, the displayed menu cell should display no text and an image.
 
-The `AddCommand` and `AddSubmenu` RPCs are going to need to be updated to account for these new features:
-
-```xml
-<function name="AddCommand" functionID="AddCommandID" messagetype="request" since="1.0">
-    <param name="cmdID" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="true" />
-    <param name="menuParams" type="MenuParams" mandatory="false" />
-    <param name="vrCommands" type="String" minsize="1" maxsize="100" maxlength="99" array="true" mandatory="false" />
-    <param name="cmdIcon" type="Image" mandatory="false" since="2.0" />
+    <param name="supportsDynamicSubMenus" type="Bool" mandatory="false">
+        <description>If true, the head unit supports dynamic sub-menus by sending OnUpdateSubMenu notifications. If true, you should not send AddCommands that attach to a parentID for an AddSubMenu until OnUpdateSubMenu is received with the menuID. At that point, you should send all AddCommands with a parentID that match the menuID. If not set, assume false.</description>
+    </param>
+</struct>
 
+<struct name="WindowCapability" since="6.0">
     <!-- New Parameters -->
-    <param name="hasIcon" type="Bool" mandatory="false" since="x.x">
-        <description>If not set, assume false. If set, then the AddCommand has an icon that has not been set in cmdIcon but can be set if the module requests with UpdateCommand.</description>
-    </param>
-</function>
+    <param name="dynamicUpdateCapabilities" type="DynamicUpdateCapabilities" mandatory="false" />
```

Comment:
Do you need `since="X.X"` here?  Also, I think you're missing this description:
><description>Contains capabilities for dynamic updating features such as image and sub-menu requests from the head unit.</description>

Reply:
Fixed!